src/Arcus.Observability.Telemetry.Sql/Extensions/ILoggerExtensions.cs

```diff
@@ -32,21 +40,62 @@ public static void LogSqlDependency(this ILogger logger, string connectionString
+        }
+
+        /// <summary>
+        /// Logs a SQL dependency
         /// </summary>
-        /// <param name="logger">Logger to use</param>
-        /// <param name="connectionString">SQL connection string</param>
-        /// <param name="tableName">Name of table</param>	
-        /// <param name="operationName">Name of the operation that was performed</param>
-        /// <param name="startTime">Point in time when the interaction with the HTTP dependency was started</param>
-        /// <param name="duration">Duration of the operation</param>
-        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
-        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
-        public static void LogSqlDependency(this ILogger logger, string connectionString, string tableName, string operationName, DateTimeOffset startTime, TimeSpan duration, bool isSuccessful, Dictionary<string, object> context = null)
+        /// <param name="logger">The logger to track the SQL dependency.</param>
+        /// <param name="connectionString">The SQL connection string.</param>
+        /// <param name="tableName">The name of tracked table in the SQL database.</param>	
+        /// <param name="operationName">The name of the operation that was performed on the SQL database.</param>
+        /// <param name="startTime">The point in time when the interaction with the HTTP dependency was started</param>
+        /// <param name="duration">The duration of the operation</param>
+        /// <param name="isSuccessful">The indication whether or not the operation was successful.</param>
+        /// <param name="context">The context that provides more insights on the dependency that was measured.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="tableName"/> or <paramref name="operationName"/> is blank.</exception>
+        public static void LogSqlDependency(
+            this ILogger logger,
+            string connectionString,
+            string tableName,
+            string operationName,
+            DateTimeOffset startTime,
+            TimeSpan duration,
+            bool isSuccessful,
+            Dictionary<string, object> context = null)
         {
+            Guard.NotNull(logger, nameof(logger), "Requires a logger instance to track telemetry");
             Guard.NotNullOrEmpty(connectionString, nameof(connectionString));
-            var connection = new SqlConnectionStringBuilder(connectionString);
+            Guard.NotNullOrWhitespace(operationName, nameof(operationName), "Requires a non-blank name of the SQL operation to track a SQL dependency");
```

Comment:
Am I mistaken or did we decide a while ago that we wouldn't guard on the operation-name ?  In other words, the `operationName` is not mandatory ?

Reply:
See this discussion: https://github.com/arcus-azure/arcus.observability/discussions/265

It is about the `tableName` parameter, but I'll create another issue for this.

Reply:
> Am I mistaken or did we decide a while ago that we wouldn't guard on the operation-name ? In other words, the operationName is not mandatory ?

Yeah, but we still guard it now, so I was trying to be consistent here. (I added this guard, bc it was already guarding on the delegated `.LogSqlDependency` call, see: https://github.com/arcus-azure/arcus.observability/blob/612335d8a76b0455ae97380cc68d8f0b83166c67/src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs#L1780)

Reply:
I've created a separate issue, so this can then be removed in the PR that tackles that issue.